packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -4421,7 +4422,12 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
                   partition={partition}
                   onChange={this.handlePartitionChange}
                   onFetchAll={this.handlePartitionFetchAll}
-                  onAppend={this.handlePartitionAppend}
+                  onAppend={
+                    onPartitionAppend !==
```

Comment:
Instead of this check, I'd remove the default prop and just check that `onPartitionAppend` is undefined.

Reply:
I'm actually getting compiler warnings now since I can't pass in undefined into onAppend due to incompatible types. Should I resolve this by changing the type of onAppend in the props interface or by passing in something like `() => undefined` instead? Alternatively, is there a third solution that is better than both of these?

Reply:
Change the prop in the interface to be optional, e.g.:
```
  onPartitionAppend?: (partitionColumn: Column, value: string) => void;
```